master/buildbot/changes/github.py

```diff
@@ -0,0 +1,259 @@
+                    pullrequest_filter=True,
+                    token=None,
+                    encoding='utf-8',
+                    pollAtLaunch=False,
+                    **kwargs):
+        base.ReconfigurablePollingChangeSource.checkConfig(
+            self,
+            name=self.name,
+            pollAtLaunch=pollAtLaunch,
+            pollInterval=pollInterval)
+        self.pollAtLaunch = pollAtLaunch
+        self.pollInterval = pollInterval
+
+    @defer.inlineCallbacks
+    def reconfigService(self,
+                        owner,
+                        repo,
+                        branches=None,
+                        pollInterval=10 * 60,
+                        useTimestamps=True,
+                        category=None,
+                        baseURL=None,
+                        project='',
+                        pullrequest_filter=True,
+                        token=None,
+                        encoding='utf-8',
+                        pollAtLaunch=False,
+                        **kwargs):
+        yield base.ReconfigurablePollingChangeSource.reconfigService(
+            self, name=self.name, **kwargs)
+
+        if baseURL is None:
+            baseURL = HOSTED_BASE_URL
+        if baseURL.endswith('/'):
+            baseURL = baseURL[:-1]
+
+        http_headers = {'User-Agent': 'Buildbot'}
+        if token is not None:
+            http_headers.update({'Authorization': 'token ' + token})
+
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, baseURL, headers=http_headers)
+
+        if not branches:
+            branches = ['master']
+
+        self.token = token
+        self.owner = owner
+        self.repo = repo
+        self.branches = branches
+        self.project = project
+        self.encoding = encoding
+        self.pollInterval = pollInterval
+
+        if callable(pullrequest_filter):
+            self.pullrequest_filter = pullrequest_filter
+        elif pullrequest_filter:
+            self.pullrequest_filter = (lambda _: pullrequest_filter)
+        else:
+            self.pullrequest_filter = None
+
+        self.lastChange = time.time()
+        self.lastPoll = time.time()
+        self.useTimestamps = useTimestamps
+        self.category = category if callable(category) else ascii2unicode(
+            category)
+        self.project = ascii2unicode(project)
+
+    def describe(self):
+        return "GitHubPullrequestPoller watching the "\
+            "GitHub repository %s/%s" % (
+                self.owner, self.repo)
+
+    @defer.inlineCallbacks
+    def _getPulls(self):
+        self.lastPoll = time.time()
+        log.debug("GitHubPullrequestPoller: polling "
+                  "GitHub repository %s/%s, branches: %s" %
+                  (self.owner, self.repo, self.branches))
+        result = yield self._http.get(
+            '/'.join(['/repos', self.owner, self.repo, 'pulls']),
+            timeout=self.pollInterval)
```

Comment:
where does this timeout comes from? I think we can drop all those timeouts.
certainly not use the same value as pollInterval. maybe another configurable constant.

In my experience, increasing timeouts is putting dust under the carpet and will bite much harder later.